core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -127,8 +147,29 @@ protected void validate(TableMetadata base) {
       }
     }
 
-    if (conflictDetectionFilter != null && base.currentSnapshot() != null) {
-      validateAddedDataFiles(base, startingSnapshotId, conflictDetectionFilter, caseSensitive);
+
+    if (validateNewDataFiles && base.currentSnapshot() != null) {
```

Comment:
Isn't the `base.currentSnapshot()` check done in `validateAddedDataFiles`? If so, we can remove it here.

Reply:
Yep, updated.